frontend/src/utils/helpers.js

```diff
@@ -95,3 +95,46 @@ export const getFiscalYear = () => {
 };
 
 export const formatParamStringToArray = (param) => (param ? param.split(",").filter((x) => x) : []);
+
+// This method sorts codes of the for '#.#.# - Lorem Ipsum'
+// where the number of integers is variable and the text is optional
+// TODO: In order to sort incidents this will need to be modified to deal with
+// parentheses
+export const compareCodes = (a, b) => {
+  // Returns the first match that is non-null.
+  const regexParse = (input) =>
+    input.match(/(?<first>[0-9]+)\.(?<second>[0-9]+)\.(?<third>[0-9]+)/) ||
+    input.match(/(?<first>[0-9]+)\.(?<second>[0-9]+)/) ||
+    input.match(/(?<first>[0-9]+)/);
+  const aCodes = regexParse(a);
+  const bCodes = regexParse(b);
+  if (!aCodes) return -1;
+  if (!bCodes) return 1;
+  let aInt;
+  let bInt;
+  const k = Math.min(aCodes.length, bCodes.length);
+  // Compares then non-null parts of two strings of potentially different lengths e.g 1.11 and 1.4.12
+  for (let i = 1; i < k; i += 1) {
+    aInt = parseInt(aCodes[i], 10);
+    bInt = parseInt(bCodes[i], 10);
+    if (aInt > bInt) return 1;
+    if (aInt < bInt) return -1;
+  }
+  // If the previous loop did not return anything and the codes are the same length then they are the same code.
+  if (aCodes.length === bCodes.length) {
+    return 0;
+  }
+  // The shorter of two codes with otherwise matching numbers is ordered first e.g 1.11 is before 1.11.12
+  return aCodes.length < bCodes.length ? -1 : 1;
```

Comment:
Nothing here actually leverages the regex very much, so one of those should be changed. Either don't use the regex at all or reference the named groups in the ordering logic.

The comments are an improvement in readability, but I'm always a bit leery about comments that describe this much implementation, since implementation can change without comments being changed, which makes things more confusing.

Reply:
I simplified the regex as much as possible.  I'm in favour of keeping the regex, mostly because generalizing to other code types is much easier to do.  i.e. 1.12.3.(3)

Reply:
I also rewrote the comments slightly, to be less verbose.